crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -180,4 +194,265 @@ impl<'a> Compressor<'a> {
 
         None
     }
+
+    fn try_fold_unary_operator<'b>(
+        &mut self,
+        unary_expr: &'b mut UnaryExpression<'a>,
+    ) -> Option<Expression<'a>> {
+        // TODO: I want compress children first, so we can fold expression like `- - 4`
+        // But our ast will visit more than once, is there better way?
+        self.fold_expression(&mut unary_expr.argument);
```

Comment:
It seems like that we need to fold many times?

Reply:
I am sorry, maybe I was wrong. I was thinking of some cascade unary expression with no side effects. But haven't found an exact example.